lib/Service/OpenProjectAPIService.php

```diff
@@ -704,53 +705,82 @@ public function getNode($userId, $fileId) {
 	}
 
 	/**
-	 * @throws \OCP\Files\InvalidPathException
+	 *
+	 * @param array<mixed> $values
+	 * @param string $userId
+	 *
+	 * @return array<int>
 	 * @throws NotFoundException
 	 * @throws \OCP\PreConditionNotMetException
 	 * @throws NotPermittedException
 	 * @throws OpenprojectErrorException
 	 * @throws \OC\User\NoUserException
 	 * @throws OpenprojectResponseException
-	 * @return int
+	 * @throws InvalidArgumentException
+	 * @throws InvalidPathException
+	 *
 	 */
 	public function linkWorkPackageToFile(
-		int $workpackageId,
-		int $fileId,
-		string $fileName,
+		array $values,
 		string $userId
-	): int {
-		$file = $this->getNode($userId, $fileId);
-		if (!$file->isReadable()) {
-			throw new NotPermittedException();
+	): array {
+		$allowedKeys = [
+			'workpackageId',
+			'fileinfo'
+		];
+		foreach ($values as $key => $value) {
+			if (!in_array($key, $allowedKeys)) {
+				throw new InvalidArgumentException('invalid key');
+			}
+		}
+		if (!is_int($values['workpackageId']) || !is_array($values['fileinfo']) || empty($values['fileinfo'])) {
+			throw new InvalidArgumentException('invalid data');
+		}
+		$fileInfos = $values['fileinfo'];
+		$elements = [];
+		// multiple files can also be linked to a single work package
+		foreach ($fileInfos as $fileInfo) {
+			if (!isset($fileInfo['id']) || !isset($fileInfo['name'])) {
+				throw new InvalidArgumentException('invalid data');
+			}
+			if (!is_int($fileInfo['id']) || $fileInfo['name'] === '' || !is_string($fileInfo['name'])) {
+				throw new InvalidArgumentException('invalid data');
+			}
```

Comment:
can't we calculate them in a single `if` statement and should these be `&&` operations?
```suggestion
			if (!isset($fileInfo['id']) && !isset($fileInfo['name']) !is_int($fileInfo['id']) && $fileInfo['name'] === '' |&& !is_string($fileInfo['name'])) {
				throw new InvalidArgumentException('invalid data');
			}
```